oak_functions_launcher/src/main.rs

```diff
@@ -91,95 +74,21 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
-
-            let mut line = String::new();
-            while reader.read_line(&mut line).expect("couldn't read line") > 0 {
-                log::info!("console: {:?}", line);
-                line.clear();
-            }
-        });
-
-        console_writer
-    };
-
-    let mut launched_instance: Box<dyn LaunchedInstance> = match cli.mode {
-        Mode::Virtual(params) => Box::new(virtualized::Instance::start(params, logs_console)?),
-        Mode::Native(params) => Box::new(native::Instance::start(params)?),
-    };
-
-    let comms = launched_instance.create_comms_channel().await?;
-    let connector_handle = channel::Connector::spawn(comms);
-
-    {
-        let mut client = schema::OakFunctionsAsyncClient::new(connector_handle.clone());
-
-        // Block for [invariant that lookup data is fully loaded](https://github.com/project-oak/oak/tree/main/oak_functions/lookup/README.md#invariant-fully-loaded-lookup-data)
-        let lookup_data =
-            lookup::load_lookup_data(&cli.lookup_data).expect("couldn't load lookup data");
-        let encoded_lookup_data =
-            lookup::encode_lookup_data(lookup_data).expect("couldn't encode lookup data");
-
-        if let Err(err) = client.update_lookup_data(&encoded_lookup_data).await {
-            panic!("couldn't send lookup data: {:?}", err)
-        }
-
-        // Spawn task to periodically refresh lookup data.
-        tokio::spawn(async move {
-            let mut interval =
-                tokio::time::interval(std::time::Duration::from_millis(1000 * 60 * 10));
-            loop {
-                // Wait before updating because we just loaded the lookup data.
-                interval.tick().await;
-
-                let lookup_data =
-                    lookup::load_lookup_data(&cli.lookup_data).expect("couldn't load lookup data");
-                let encoded_lookup_data =
-                    lookup::encode_lookup_data(lookup_data).expect("couldn't encode lookup data");
-
-                if let Err(err) = client.update_lookup_data(&encoded_lookup_data).await {
-                    panic!("couldn't send lookup data: {:?}", err)
-                }
-            }
-        });
-    }
-
-    let wasm_bytes = fs::read(&cli.wasm)
-        .with_context(|| format!("couldn't read Wasm file {}", &cli.wasm.display()))
-        .unwrap();
-    log::info!(
-        "read Wasm file from disk {} ({} bytes)",
-        &cli.wasm.display(),
-        wasm_bytes.len()
-    );
-
-    let request = schema::InitializeRequest {
-        wasm_module: wasm_bytes,
-        constant_response_size: cli.constant_response_size,
-    };
-
-    let mut client = schema::OakFunctionsAsyncClient::new(connector_handle.clone());
-    let result = client
-        .initialize(&request)
-        .await
-        .flatten()
-        .expect("couldn't initialize the service");
+    let (mut launched_instance, connector_handle, result) = oak_functions_launcher::create(
```

Comment:
Maybe don't use the name `result` here, as it confused me for a moment (I was thinking that it was a `Result<>`, and if it indeed was, why don't you unwrap it with a `?`?)